dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/MemoryAllocation.java

```diff
@@ -81,6 +81,10 @@ private void createInitEvents(Program program) {
 
         int nextThreadId = threads.get(threads.size() - 1).getId() + 1;
         for(MemoryObject memObj : program.getMemory().getObjects()) {
+            // We do not create Init events for virtual addresses
+            if(memObj.isVirtual()) {
+                continue;
+            }
```

Comment:
A method `memory.getPhysicalObjects()` would be nice. Together with the aforementioned `memory.allocateVirtual`, the `Memory` class provides a good interface to properly separate physical and virtual addresses.

Reply:
This special case has been removed in the new commit